eng/packages/http-client-csharp-mgmt/emitter/src/resource-detection.ts

```diff
@@ -190,20 +190,16 @@ export async function updateClients(
     }
   }
 
-  // the last step, add the decorator to the resource model
-  for (const model of resourceModels) {
-    const metadata = resourceModelToMetadataMap.get(
-      model.crossLanguageDefinitionId
-    );
-    if (metadata) {
-      addResourceMetadata(sdkContext, model, metadata);
-    }
-  }
-  // and add the methodMetadata decorator to the non-resource methods
-  addNonResourceMethodDecorators(
-    codeModel,
-    Array.from(nonResourceMethods.values())
+  // Build the unified ARM provider schema
+  const armProviderSchema = buildArmProviderSchema(
+    resourceModels,
+    resourceModelToMetadataMap,
+    nonResourceMethods,
+    sdkContext
   );
```

Comment:
I think everything above this part is also "building the ARM provider schema related information".
could you extract them as a function as well, and call it inside `buildArmProviderSchema`?